test/unittest/unittest_common_properties.cpp

```diff
@@ -101,92 +101,49 @@ TEST(NameProperty, mustStartWithAlphaNumeric_01_n) {
   EXPECT_THROW(n.set("+layer"), std::invalid_argument);
 }
 
-TEST(InputSpecProperty, setPropertyValid_p) {
+TEST(InputConnection, setPropertyValid_p) {
   using namespace nntrainer::props;
   {
-    InputSpec expected(
-      ConnectionSpec({Name("A"), Name("B"), Name("C")}, "concat"));
+    InputConnection expected(Connection("a", 0));
 
-    InputSpec actual;
-    nntrainer::from_string("A, B, C", actual);
+    InputConnection actual;
+    nntrainer::from_string("A", actual);
     EXPECT_EQ(actual, expected);
-    EXPECT_EQ("a,b,c", nntrainer::to_string(actual));
+    EXPECT_EQ("a(0)", nntrainer::to_string(actual));
   }
 
   {
-    InputSpec expected(
-      ConnectionSpec({Name("A"), Name("B"), Name("C")}, "addition"));
-
-    InputSpec actual;
-    nntrainer::from_string("A+ B +C", actual);
-    EXPECT_EQ("a+b+c", nntrainer::to_string(actual));
-
-    EXPECT_EQ(actual, expected);
-  }
-
-  {
-    InputSpec expected(ConnectionSpec({Name("a")}, ConnectionSpec::NoneType));
-
-    InputSpec actual;
-    nntrainer::from_string("a", actual);
-    EXPECT_EQ("a", nntrainer::to_string(actual));
+    InputConnection expected(Connection("a", 2));
 
+    InputConnection actual;
+    nntrainer::from_string("a(2)", actual);
     EXPECT_EQ(actual, expected);
+    EXPECT_EQ("a(2)", nntrainer::to_string(actual));
   }
 }
 
-TEST(InputSpecProperty, emptyString_n_01) {
+TEST(InputConnection, emptyString_n_01) {
   using namespace nntrainer::props;
-  InputSpec actual;
+  InputConnection actual;
   EXPECT_THROW(nntrainer::from_string("", actual), std::invalid_argument);
 }
 
-TEST(InputSpecProperty, combinedOperator_n_01) {
-  using namespace nntrainer::props;
-  InputSpec actual;
-  EXPECT_THROW(nntrainer::from_string("A,B+C", actual), std::invalid_argument);
-}
-
-TEST(InputSpecProperty, combinedOperator_n_02) {
-  using namespace nntrainer::props;
-  InputSpec actual;
-  EXPECT_THROW(nntrainer::from_string("A+B,C", actual), std::invalid_argument);
-}
-
-TEST(InputSpecProperty, noOperator_n_01) {
-  using namespace nntrainer::props;
-  InputSpec actual;
-  EXPECT_THROW(nntrainer::from_string("A B", actual), std::invalid_argument);
-}
-
-TEST(InputSpecProperty, noOperator_n_02) {
-  using namespace nntrainer::props;
-  InputSpec actual;
-  EXPECT_THROW(nntrainer::from_string("A B", actual), std::invalid_argument);
-}
-
-TEST(InputSpecProperty, leadingOperator_n_01) {
-  using namespace nntrainer::props;
-  InputSpec actual;
-  EXPECT_THROW(nntrainer::from_string(",A,B", actual), std::invalid_argument);
-}
-
-TEST(InputSpecProperty, leadingOperator_n_02) {
+TEST(InputConnection, onlyIndex_n_01) {
   using namespace nntrainer::props;
-  InputSpec actual;
-  EXPECT_THROW(nntrainer::from_string("+A+B", actual), std::invalid_argument);
+  InputConnection actual;
+  EXPECT_THROW(nntrainer::from_string("[0]", actual), std::invalid_argument);
```

Comment:
shouldn't the brackets be `()` instead of `[]`. Using `[]` is also a negative test case but might give error because of different reason.

Reply:
right, I'll change :)

Reply:
I'll handle this with https://github.com/nnstreamer/nntrainer/pull/1732#discussion_r758029310 :)

Reply:
sure.